Packet++/header/VrrpLayer.h

```diff
@@ -0,0 +1,545 @@
+		 * @return The virtual IP addresses in this message
+		 */
+		std::vector<IPAddress> getIPAddresses() const;
+
+		/**
+		 * A method that gets first VRRP virtual IP address
+		 * @return A pointer to the first virtual IP address or NULL if no virtual IP address exist. Notice the return value is a pointer to the real data,
+		 * so changes in the return value will affect the packet data
+		 */
+		uint8_t *getFirstIPAddress() const;
+
+		/**
+		 * Get the virtual IP address that comes next to a given virtual IP address. If "ipAddress" is NULL then NULL will be returned.
+		 * If "ipAddress" is the last virtual IP address or if it is out of layer bounds NULL will be returned also. Notice the return value is a
+		 * pointer to the real data casted to IPAddress type (as opposed to a copy of the option data). So changes in the return
+		 * value will affect the packet data
+		 * @param[in] ipAddress The virtual IP address to start searching from
+		 * @return The next virtual IP address or NULL if "ipAddress" is NULL, last or out of layer bounds
+		 */
+		uint8_t *getNextIPAddress(uint8_t *ipAddress) const;
+
+		/**
+		 * Add virtual IP address at a the end of the virtual IP address list. The vrrp_packet#ipAddressCount field will be
+		 * incremented accordingly
+		 * @param[in] ipAddresses A vector containing all the virtual IP address
+		 * @return ture if add successfully, false otherwise.
+		 */
+		bool addIPAddresses(const std::vector<IPAddress> &ipAddresses);
+
+		/**
+		 * Add a new virtual IP address at a the end of the virtual IP address list. The vrrp_packet#ipAddressCount field will be
+		 * incremented accordingly
+		 * @param[in] ipAddress A vector containing all the virtual IP address
+		 * @return ture if add successfully, false otherwise.
+		 */
+		bool addIPAddress(IPAddress &ipAddress);
+
+		/**
+		 * Remove a virtual IP address at a certain index. The vrrp_packet#ipAddressCount field will be decremented accordingly
+		 * @param[in] index The index of the virtual IP address to be removed
+		 * @return True if virtual IP address was removed successfully or false otherwise. If false is returned an appropriate error message
+		 * will be printed to log
+		 */
+		bool removeIPAddressAtIndex(int index);
+
+		/**
+		 * Remove all virtual IP addresses in the message. The vrrp_packet#ipAddressCount field will be set to 0
+		 * @return True if virtual IP addresses were cleared successfully or false otherwise. If false is returned an appropriate error message
+		 * will be printed to log
+		 */
+		bool removeAllIPAddresses();
+
+		/**
+		 * Copy IP address to data
+		 * @param[in] data A data pointer
+		 * @param[in] ipAddress IP address
+		 */
+		void copyIPAddressToData(uint8_t *data, const IPAddress &ipAddress) const;
+
+		/**
+		 * Get IP address from data
+		 * @return true if get successfully, otherwise return false.
+		 */
+		bool getIPAddressFromData(uint8_t *data, IPAddress &ipAddress) const;
```

Comment:
This method should be private

Reply:
Fixed in https://github.com/seladb/PcapPlusPlus/pull/1118/commits/0233d684d8aff42029b747845745d835aa066beb